lambda/src/lib.rs

```diff
@@ -0,0 +1,202 @@
+//! Anonymiser lambda
+//!
+//! This lambda is used to convert incoming scripts from the TRE production bucket into anonymised packages
+//!
+//! Given the following input:
+//! ```json
+//! {
+//!   "parameters": {
+//!     "s3Bucket": "input-bucket",
+//!     "s3Key": "TRE-TDR-2023-ABC.tar.gz"
+//!   }
+//! }
+//! ```
+//! The lambda will:
+//! * Download the file from S3 to local disk
+//! * Anonymise it using the anonymise library
+//! * Upload it to S3 using the `OUTPUT_BUCKET` environment variable
+//! * Send the SQS message to the queue specified in the `OUTPUT_QUEUE` environment variable
+
+use anonymise::process_package;
+use aws_config::meta::region::RegionProviderChain;
+use aws_config::{BehaviorVersion, SdkConfig};
+use aws_lambda_events::sqs::SqsMessage;
+use aws_sdk_s3::primitives::ByteStream;
+use aws_sdk_s3::Client as S3Client;
+use aws_sdk_sqs::Client as SQSClient;
+use lambda_runtime::Error;
+use serde::{Deserialize, Serialize};
+use std::fs;
+use std::fs::File;
+use std::io::Write;
+use std::path::PathBuf;
+
+/// The bucket and key for the file we are processing
+#[derive(Deserialize, Serialize)]
+struct MessageBody {
+    parameters: S3Details,
+}
+
+#[derive(Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+struct S3Details {
+    status: String,
+    reference: String,
+    s3_bucket: String,
+    s3_key: String,
+}
+
+/// # Processes the SQS message.
+///
+/// This will download the file specified in the message body, anonymise it, upload it to S3 and send the message on to the output queue.
+pub async fn process_record(
+    message: &SqsMessage,
+    working_directory: PathBuf,
+    s3_endpoint_url: Option<&str>,
+    sqs_endpoint_url: Option<&str>,
+) -> Result<PathBuf, Error> {
+    let body = message
+        .body
+        .as_ref()
+        .ok_or("No body found in the SQS message")?;
+    let s3_client = create_s3_client(s3_endpoint_url).await;
+    let sqs_client = create_sqs_client(sqs_endpoint_url).await;
+
+    let message_body: MessageBody = serde_json::from_str(&body)?;
+    let parameters = message_body.parameters;
+    let input_file_path = download(
+        &s3_client,
+        parameters.s3_bucket,
+        parameters.s3_key,
+        &working_directory,
+    )
+    .await?;
+    let output_path = &working_directory.join(PathBuf::from("output"));
+    fs::create_dir_all(output_path)?;
+    let output_tar_path = process_package(output_path, &input_file_path)?;
+    let file_name = output_tar_path
+        .file_name()
+        .and_then(|oss| oss.to_str())
+        .expect("Cannot parse file name from output path");
+
+    let output_bucket = std::env::var("OUTPUT_BUCKET")?;
+    upload(&s3_client, &output_tar_path, &output_bucket, &file_name).await?;
+
+    let output_queue = std::env::var("OUTPUT_QUEUE")?;
+    let reference = parameters.reference;
```

Comment:
Maybe `tre_reference`?

Reply:
I think it's ok as is. It's called reference in the json.